main_test.go

```diff
@@ -0,0 +1,195 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"gitlab.com/NebulousLabs/errors"
+)
+
+// TestLoadDBCredentials is a unit test that covers the loadDBCredentials
+// helper.
+func TestLoadDBCredentials(t *testing.T) {
+	t.Parallel()
+
+	variables := []string{
+		"SKYNET_DB_USER",
+		"SKYNET_DB_PASS",
+		"SKYNET_DB_HOST",
+		"SKYNET_DB_PORT",
+	}
+
+	// create a function to restore the environment
+	restoreEnvFn := restoreEnv(variables)
+	defer func() {
+		err := restoreEnvFn()
+		if err != nil {
+			t.Error(err)
+		}
+	}()
+
+	// set every env variable to its name
+	for _, variable := range variables {
+		os.Setenv(variable, variable)
+	}
+
+	// load db credentials and assert its output (happy case)
+	connstring, credentials, err := loadDBCredentials()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if credentials.Username != "SKYNET_DB_USER" || credentials.Password != "SKYNET_DB_PASS" {
+		t.Fatal("unexpected", credentials)
+	}
+	if connstring != "mongodb://SKYNET_DB_HOST:SKYNET_DB_PORT" {
+		t.Fatal("unexpected", connstring)
+	}
+
+	// unset every env variable one by one and assert the helper indicates what
+	// environment variable is missing
+	for _, variable := range variables {
+		bkp := os.Getenv(variable)
+		err = os.Unsetenv(variable)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		_, _, err := loadDBCredentials()
+		if err == nil || !strings.Contains(err.Error(), fmt.Sprintf("missing env var %v", variable)) {
+			t.Fatal("unexpected outcome", err)
+		}
+
+		// put it back
+		err = os.Setenv(variable, bkp)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+}
+
+// TestLoadEmailCredentials is a unit test that covers the loadEmailCredentials
+// helper.
+func TestLoadEmailCredentials(t *testing.T) {
+	t.Parallel()
+
+	variables := []string{
+		"EMAIL_SERVER",
+		"EMAIL_USERNAME",
+		"EMAIL_PASSWORD",
+	}
+
+	// create a function to restore the environment
+	restoreEnvFn := restoreEnv(variables)
```

Comment:
Do you need this? Are environment variables even persisted between tests? I'd double check that and if not we can clean this up a bit. 

Otherwise I'd still consider using `t.Cleanup(restoreEnv(variables))` to save a few lines.

Reply:
I didn't know about `t.Cleanup`, that's nice. I can't use it here though because the function signatures don't match. 

This `restoreEnv` thing is mainly going off of a unit test Ivo once wrote to verify `loadDBCredentials`. I personally think it's a bit of an overkill but at the same time, I don't see a good reason to remove it either. I'll cc @ro-tex on this one I think, but I kind of want to just resolve this comment and keep it, we only add these lines for unit tests that involve loading stuff from the environment so I don't see why it's a big deal.

Reply:
I don't remember that `loadDBCredentials` test. I don't feel partial to keeping it - remove it if it doesn't seem necessary.

Re `t.Cleanup`, I prefer it over the `restoreEnv` defer.

In any case, any test that calls `os.Setenv()` should not be `t.Parallel()` - this is begging for an NDF.

Reply:
Removed `t.Parallel`, I vote we keep the rest to be honest, can't use `t.Cleanup` because function sig. does not match

Reply:
I don't think `t.Cleanup` vs `restoreEnv` is a big deal. I'm fine with keeping it as it is.